### PR TITLE
adaption to new behaviour of darklak + refactoring

### PR DESCRIPTION
**Pull request by akasolace**
1. changes proposed in this pull request:
 
   - fixes issue #771
   
   - if not fixing an existing issue, comments explaining the purpose of the PR should be provided)
 
 
2. changelog and release_notes ...

 - [ ] have been updated
 - [x] do not require update

3. [Optional] suggested person to review this PR @weisJ

### CODE REVIEW

File: src/main/java/core/gui/model/SpielerCBItem.java [hunk 1]
```diff
@@ -51,16 +52,14 @@ public SpielerCBItem(String text, float poswert, Player player, boolean useCusto
...
         if (player != null) {
             m_clEntry.updateComponent(player, HOVerwaltung.instance().getModel().getLineupWithoutRatingRecalc()
                             .getPositionBySpielerId(player.getSpielerID()), getPositionsBewertung(), m_bAlternativePosition,
                     m_sText);
 
-            if (m_bMultiLine) {
-                m_clEntry.getComponent(isSelected).setPreferredSize(new Dimension(m_clEntry.getComponent(isSelected).getPreferredSize().width, PLAYER_COMBO_HEIGHT));
-            }
-            return m_clEntry.getComponent(isSelected);
+            JComponent comp = m_clEntry.getComponent(isSelected, index==-1);
```
Comment (akasolace) [inline src/main/java/core/gui/model/SpielerCBItem.java:60]: @weisJ I just saw that you did commit https://github.com/weisJ/darklaf/commit/a9e0ce3a220f4c56027d5f28d1fc70a0a022a3f6

does it mean you reverted to old behaviour and what I did in the PR will eventually not be necessery ?

Comment (weisJ) [reply src/main/java/core/gui/model/SpielerCBItem.java:60]: The change was made to make the `index == -1` check functional. The changes are still necessary.

File: src/main/java/core/gui/model/SpielerCBItem.java [hunk 2]
```diff
@@ -51,16 +52,14 @@ public SpielerCBItem(String text, float poswert, Player player, boolean useCusto
...
+            if (m_bMultiLine) comp.setPreferredSize(new Dimension(comp.getPreferredSize().width, PLAYER_COMBO_HEIGHT));
```
[the remark this answers is not in the document]
Comment (akasolace) [reply src/main/java/core/gui/model/SpielerCBItem.java:61]: I tried without that and the rendering is not good ... same height as singleLine combobox ....

[the remark this answers is not in the document]
Comment (weisJ) [reply src/main/java/core/gui/model/SpielerCBItem.java:61]: Did you make sure to remove any calls to `setPreferredSize` to the component e.g in `SpielerLabelEntry#createComponent` a preferred size is set.

[the remark this answers is not in the document]
Comment (akasolace) [reply src/main/java/core/gui/model/SpielerCBItem.java:61]: you are correct, I just tried again. This time works really fine without any `setPreferredSize`I should have missed on or two the previous time.

[the remark this answers is not in the document]
Comment (akasolace) [reply src/main/java/core/gui/model/SpielerCBItem.java]: actually, just realized I can replace the one above by `m_jbFlipSide.setMargin(new Insets(0, 0, 0, 0));` (I thought this was the default behaviour) and it works well. No more call to `setPreferredSize` at all, it makes things much more easy !

**Review by wsbrenk — APPROVED** [review]
looks good to me - but i'm no ui expert. let's test the dev version.

**Comment by wsbrenk** [comment]
compare new and old view of the lineup page:

image

**Comment by wsbrenk** [comment]
the different widths of the player boxes make the new view very restless (nimbus). the group information (A-F) is no longer displayed - was this intentional?

**Pull request merged**